3/changelog alpha 7 (#2892)
apostrophecms/apostrophe: CHANGELOG.md

# Changelog

## 3.0.0-alpha.7 - 2021-04-07

### Breaks

* The `trash` property has been renamed `archived`, and throughout the UI we refer to "archiving" and the "archive" rather than "move to trash" and the "trash can." A database migration is included to address this for existing databases. However, **if you set the minimumPark option, or used a boilerplate in which it is set,** you will need to **change the settings for the `parkedId: 'trash'` page to match those [currently found in the `minimumPark` option setting in the `@apostrophecms/page` source code](https://github.com/apostrophecms/apostrophe/blob/481252f9bd8f42b62648a0695105e6e9250810d3/modules/%40apostrophecms/page/index.js#L25-L32).

### Adds

* General UX and UI improvements to the experience of moving documents to and from the archive, formerly known as the trash.
* The `className` option can now be passed to the core widgets via page templates. If present this takes precedence over a `className` set at the module level.
* Links to each piece are available in the manage view when appropriate.

### Fixes

> * Bug fix: don't update the modification timestamp of a document simply because of an advisory lock, as the user might decide not to save any actual edits.

Reviewer: ```suggestion
* Advisory lock no longer triggers an update to the modification timestamp of a document.
```

> * Prevent content loss by blocking attempts to connect Apostrophe 3.x to an Apostrophe 2.x database. Content migration tools are planned of course.

Reviewer: ```suggestion
* Attempts to connect Apostrophe 3.x to an Apostrophe 2.x database are blocked to prevent content loss
```

> * You can always save a draft of a new document, wherever the concept is relevant.

Reviewer: ```suggestion
* "Save as Draft" is now available as soon as a new document is created
```

> * Areas nested in array schema fields can now be edited in context on the page.

Reviewer: ```suggestion
* Areas nested in array schema fields can now be edited in context.
```

* When using `apos.image.first`, the alt attribute of the image piece is available on the returned attachment object as `._alt`. In addition, `_credit` and `_creditUrl` are available.

> * You can now set core widget `className` options in area widget configuration, not only in widget module options.

Reviewer: ```suggestion
* You can now set core widget `className` options in area widget configuration.
```

Reviewer: wouldn't this be an addition as well? I think you actually say that above, feels duplicative.

> * Do not immediately redirect to new pages and pieces.

Reviewer: This wouldn't be a `fix` because it wasn't a bug. It would be an `add` or a `change`, yeah?

> * Do not immediately redirect to new pages and pieces.

Reviewer: ```suggestion
* Removes the automatic redirect for new pieces and pages
```

### Breaks

* The `trash` property has been renamed `archived`, and throughout the UI we refer to "archiving" and the "archive" rather than "move to trash" and the "trash can." A database migration is included to address this for existing databases. However, **if you set the minimumPark option, or used a boilerplate in which it is set,** you will need to **change the settings for the `parkedId: 'trash'` page to match those [currently found in the `minimumPark` option setting in the `@apostrophecms/page` source code](https://github.com/apostrophecms/apostrophe/blob/481252f9bd8f42b62648a0695105e6e9250810d3/modules/%40apostrophecms/page/index.js#L25-L32).

### Adds

* General UX and UI improvements to the experience of moving documents to and from the archive, formerly known as the trash.
* Links to each piece are available in the manage view when appropriate.
* Search is implemented in the media library.
* You can now pass core widgets a `className` option when configuring them as part of an area.

### Changes

* Do not immediately redirect to new pages and pieces.

### Fixes

* Advisory lock no longer triggers an update to the modification timestamp of a document.

> * Attempts to connect Apostrophe 3.x to an Apostrophe 2.x database are blocked to prevent content loss

Reviewer: ```suggestion
* Attempts to connect Apostrophe 3.x to an Apostrophe 2.x database are blocked to prevent content loss.
```

> * "Save as Draft" is now available as soon as a new document is created

Reviewer: ```suggestion
* "Save as Draft" is now available as soon as a new document is created.
```
